sqlmesh/dbt/manifest.py

```diff
@@ -308,17 +310,41 @@ def _load_disabled(self) -> None:
             elif node.resource_type == "source":
                 self._disabled_sources.discard(node.name)
 
-    def _macro_source_ref_dependencies(
-        self, macros: t.List[MacroReference], default_package: str
+    def _flatten_dependencies_from_macros(
+        self,
+        macros: t.List[MacroReference],
+        default_package: str,
+        visited: t.Optional[t.Set[t.Tuple[str, str]]] = None,
     ) -> Dependencies:
+        if visited is None:
+            visited = set()
+
         dependencies = Dependencies()
         for macro in macros:
-            macro_config = self._macros_per_package[macro.package or default_package].get(
+            macro_package = macro.package or default_package
+
+            if (macro_package, macro.name) in visited:
+                continue
+            visited.add((macro_package, macro.name))
+
+            macro_dependencies = self._macro_flatten_dependencies.get(macro_package, {}).get(
```

Comment:
Should we use `none` as the default for the case when the macro truly has no dependencies?

Reply:
If the dict for that *package* wasn't set yet, we fallback to `{}` to safely chain with a subsequent `get()`.

Reply:
Ahh, my eyes completely glanced over that since it wasn't a changed line. Makes sense now.